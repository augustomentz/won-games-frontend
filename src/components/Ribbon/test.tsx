import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Ribbon from '.'

describe('<Ribbon />', () => {
	it('Should render the text correctly', () => {
		renderWithTheme(<Ribbon>Bestseller</Ribbon>)

		expect(screen.getByText(/Bestseller/i)).toBeInTheDocument()
	})

	it('Should render with the primary color', () => {
		renderWithTheme(<Ribbon>Bestseller</Ribbon>)

		expect(screen.getByText(/Bestseller/i)).toHaveStyle({
			backgroundColor: '#F231A5'
		})
	})

	it('Should render with the secondary color', () => {
		renderWithTheme(<Ribbon color="secondary">Bestseller</Ribbon>)

		expect(screen.getByText(/Bestseller/i)).toHaveStyle({
			backgroundColor: '#3CD3C1'
		})
	})

	it('Should render the normal font size as default', () => {
		renderWithTheme(<Ribbon>Bestseller</Ribbon>)

		expect(screen.getByText(/Bestseller/i)).toHaveStyle({
			height: '3.6rem',
			fontSize: '1.4rem'
		})
	})

	it('Should render the small font size', () => {
		renderWithTheme(<Ribbon size="small">Bestseller</Ribbon>)

		expect(screen.getByText(/Bestseller/i)).toHaveStyle({
			height: '2.6rem',
			fontSize: '1.2rem'
		})
	})
})
