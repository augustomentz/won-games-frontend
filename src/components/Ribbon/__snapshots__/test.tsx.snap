// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon /> Should be render the heading 1`] = `
<main
  class=""
>
  <h1>
    Ribbon
  </h1>
</main>
`;
